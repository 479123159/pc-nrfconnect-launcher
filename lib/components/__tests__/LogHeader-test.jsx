/* eslint-disable import/first */

// Do not decorate components
jest.mock('../../util/plugins', () => ({
    decorate: component => component,
}));

import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import LogHeader from '../LogHeader';

describe('LogHeader', () => {
    it('should render autoScroll enabled', () => {
        expect(renderer.create(
            <LogHeader
                autoScroll
                onOpenLogFile={() => {}}
                onClearLog={() => {}}
                onToggleAutoScroll={() => {}}
            />,
        )).toMatchSnapshot();
    });

    it('should render autoScroll disabled', () => {
        expect(renderer.create(
            <LogHeader
                autoScroll={false}
                onOpenLogFile={() => {}}
                onClearLog={() => {}}
                onToggleAutoScroll={() => {}}
            />,
        )).toMatchSnapshot();
    });

    it('should invoke onOpenLogFile when openLogFileButton has been clicked', () => {
        const onOpenLogFile = jest.fn();
        const wrapper = mount(
            <LogHeader
                autoScroll
                onOpenLogFile={onOpenLogFile}
                onClearLog={() => {}}
                onToggleAutoScroll={() => {}}
                openLogFileButtonTitle="Open log file"
            />,
        );
        wrapper.find('[title="Open log file"]').simulate('click');

        expect(onOpenLogFile).toHaveBeenCalled();
    });

    it('should invoke onClearLog when clearLogButton has been clicked', () => {
        const onClearLog = jest.fn();
        const wrapper = mount(
            <LogHeader
                autoScroll
                onOpenLogFile={() => {}}
                onClearLog={onClearLog}
                onToggleAutoScroll={() => {}}
                clearLogButtonTitle="Clear log"
            />,
        );
        wrapper.find('[title="Clear log"]').simulate('click');

        expect(onClearLog).toHaveBeenCalled();
    });

    it('should invoke onToggleAutoScroll when autoScrollButton has been clicked', () => {
        const onToggleAutoScroll = jest.fn();
        const wrapper = mount(
            <LogHeader
                autoScroll
                onOpenLogFile={() => {}}
                onClearLog={() => {}}
                onToggleAutoScroll={onToggleAutoScroll}
                autoScrollButtonTitle="Toggle autoscroll"
            />,
        );
        wrapper.find('[title="Toggle autoscroll"]').simulate('click');

        expect(onToggleAutoScroll).toHaveBeenCalled();
    });
});
