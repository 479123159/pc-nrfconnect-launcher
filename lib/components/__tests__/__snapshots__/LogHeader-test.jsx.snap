exports[`LogHeader should render autoScroll disabled 1`] = `
<div
  className="log-header">
  <div
    className="log-header-text">
    Log
  </div>
  <div
    className="padded-row log-header-buttons">
    <button
      className="btn btn-primary btn-xs btn-nordic "
      onClick={[Function]}
      title="Open log file">
      <span
        aria-hidden="true"
        className="icon-doc-text" />
    </button>
    <button
      className="btn btn-primary btn-xs btn-nordic "
      onClick={[Function]}
      title="Clear log">
      <span
        aria-hidden="true"
        className="icon-trash" />
    </button>
    <button
      className="btn btn-primary btn-xs btn-nordic "
      onClick={[Function]}
      title="Scroll automatically">
      <span
        aria-hidden="true"
        className="icon-down" />
    </button>
  </div>
</div>
`;

exports[`LogHeader should render autoScroll enabled 1`] = `
<div
  className="log-header">
  <div
    className="log-header-text">
    Log
  </div>
  <div
    className="padded-row log-header-buttons">
    <button
      className="btn btn-primary btn-xs btn-nordic "
      onClick={[Function]}
      title="Open log file">
      <span
        aria-hidden="true"
        className="icon-doc-text" />
    </button>
    <button
      className="btn btn-primary btn-xs btn-nordic "
      onClick={[Function]}
      title="Clear log">
      <span
        aria-hidden="true"
        className="icon-trash" />
    </button>
    <button
      className="btn btn-primary btn-xs btn-nordic active"
      onClick={[Function]}
      title="Scroll automatically">
      <span
        aria-hidden="true"
        className="icon-down" />
    </button>
  </div>
</div>
`;
